### PR TITLE
API-579 SQL 5 docs & samples

### PR DESCRIPTION
**Pull request by srknzl**
merge #963 first

1. Updates sql docs according to new Hazelcast 5 docs
2. Updates code samples adds a new one
3. Adds try catch to code samples because of this issue #983 
4. Removes leftover hzLocalDate kind of usages

### CODE REVIEW

**Comment by srknzl** [comment]
@mdumandag FYI, I prepared this PR to be the next one to be merged

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -2190,12 +2190,42 @@ console.log(value);
...
+- [SHOW MAPPINGS:](https://docs.hazelcast.com/hazelcast/latest/sql/show-mappings.html) Get the names of existing mappings.
+- [DROP MAPPING](https://docs.hazelcast.com/hazelcast/latest/sql/drop-mapping.html) Remove a mapping.
+
+#### Job Management Statements
+
+- [CREATE JOB:](https://docs.hazelcast.com/hazelcast/latest/sql/create-job.html) Create a job that is not tied to the client session.
+- [ALTER JOB:](https://docs.hazelcast.com/hazelcast/latest/sql/alter-job.html) Restart, suspend, or resume a job.
+- [SHOW JOBS:](https://docs.hazelcast.com/hazelcast/latest/sql/show-jobs.html) Get the names of all running jobs.
+- [DROP JOB:](https://docs.hazelcast.com/hazelcast/latest/sql/drop-job.html) Cancel a job.
+- [CREATE OR REPLACE SNAPSHOT (Enterprise only):](https://docs.hazelcast.com/hazelcast/latest/sql/create-snapshot.html) Create a snapshot of a running job, so you can stop and restart it at a later date.
```
Comment (mdumandag) [inline DOCUMENTATION.md:2223]: I think stopping or restarting the job is not the main intent. The docs say 
```
you can use to start other jobs or keep as a backup
```

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -2190,12 +2190,42 @@ console.log(value);
...
+- [DROP SNAPSHOT (Enterprise only):](https://docs.hazelcast.com/hazelcast/latest/sql/drop-snapshot.html) Cancel a running job.
```
Comment (mdumandag) [inline DOCUMENTATION.md:2224]: Does it cancel the job or the snapshot?

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -2305,57 +2331,34 @@ the following query does not return the `this` field, because the value has nest
...
-| **DATE**                     | `string`            |
-| **TIME**                     | `string`            |
-| **TIMESTAMP**                | `string`            |
-| **TIMESTAMP_WITH_TIME_ZONE** | `string`            |
-| **OBJECT**                   | Any class           |
-| **NULL**                     | `null`              |
+### 8.7.3. Data Types
 
-#### Decimal String Format
+The SQL service supports a set of SQL data types. The  table below shows SQL data types and corresponding Javascript types:
```
Comment (mdumandag) [inline DOCUMENTATION.md:2336]:
```suggestion
The SQL service supports a set of SQL data types. The table below shows SQL data types and corresponding JavaScript types:
```

File: DOCUMENTATION.md (hunk 1 above)
Comment (srknzl) [reply DOCUMENTATION.md:2223]: I am not sure really what is the intent of it but I copied these from https://docs.hazelcast.com/hazelcast/latest/sql/sql-statements.html

File: DOCUMENTATION.md (hunk 2 above)
Comment (srknzl) [reply DOCUMENTATION.md:2224]: Again, copy paste, so I think we should change the Hazelcast docs if it's wrong

File: DOCUMENTATION.md (hunk 1 above)
Comment (mdumandag) [reply DOCUMENTATION.md:2223]: Oh, I see. I clicked on the individual elements. Let's leave it like this then

**Review by mdumandag — APPROVED** [review]

**Pull request merged**